src/RestClient/RestClient/src/Authentication/BasicAccessAuthenticator.cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace ClickView.Extensions.RestClient.Authentication
+{
+    using System;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Requests;
+
+    /// <summary>
+    ///     Basic Access Authenticator which adds an Authorization header value to each request
+    ///     Follows this spec: https://tools.ietf.org/html/rfc7617
+    /// </summary>
+    public class BasicAccessAuthenticator : IAuthenticator
+    {
+        private readonly string _key;
+        private readonly string _val;
+
+        public BasicAccessAuthenticator(string key, string val)
```

Comment:
the RFC defines the names as `user-id` (`userId`) and `password`. Could you use those instead of `key` and `val`

>    2.  constructs the user-pass by concatenating the user-id, a single
       colon (":") character, and the password,